src/page/sinfo.rs

```diff
@@ -32,28 +37,50 @@ impl core::fmt::Display for SecInfo {
         match self.class {
             Class::Secs => write!(f, "S"),
             Class::Tcs => write!(f, "T"),
-            Class::Reg => write!(f, "{}", self.flags),
-            Class::Va => write!(f, "V"),
-            Class::Trim => write!(f, "^"),
+            Class::Regular => write!(f, "{}", self.flags),
+            Class::VersionArray => write!(f, "V"),
+            Class::Trimmed => write!(f, "^"),
+            Class::ShadowStackFirst => write!(f, "F"),
+            Class::ShadowStackRest => write!(f, "R"),
         }
     }
 }
 
-impl SecInfo {
-    /// Creates a `SecInfo` instance for regular pages
-    pub const fn reg(flags: Flags) -> Self {
-        Self {
-            flags,
-            class: Class::Reg,
-            reserved: [0; 31],
-        }
+impl From<Class> for SecInfo {
+    fn from(class: Class) -> Self {
+        SecInfo::new(class, None)
     }
+}
+
+/// Error codes for `SecInfo::accept()`
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub enum AcceptError {
+    /// CPU cores have not exited from the previous grace period.
+    PageNotTracked,
+    /// Attributes of the destination page are incorrect.
+    PageAttributesMismatch,
+}
+
+/// Error codes for `SecInfo::accept_copy()`
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub enum AcceptCopyError {
+    /// Attributes of the destination page are incorrect.
+    PageAttributesMismatch,
+}
```

Comment:
You forgot to combine these error types into a single type.

Reply:
So the thing is that `EACCEPTCOPY` does not return `PageNotTracked`, so this documents with the Rust's type system this fact.

Reply:
`EACCEPTCOPY` can be only paired with `EAUG`. It works neither with `EMODPR` nor `EMODT`.

Reply:
I'm not going to strongly oppose combining them but still this gives IMHO better documentation of their use and purpose.

Reply:
@bstrie What do you think? :-)

Reply:
Hm, I think the difference here comes down to taste. Error enums in Rust aren't always fine-grained, which is to say that it's fairly common for functions to return an error enum where that function isn't capable of producing *all* the possible errors that the enum can represent. However, it's not that there's anything wrong with having fine-grained enums, it's just that it's often a pain to define unique error enums for each function. If, as you say, this is representing a useful property of the API, then I'd be inclined to leave it as long as it isn't imposing a maintenance burden elsewhere.

Reply:
OK I'll just merge these, as said this not something that I'm strongly opposing.

Reply:
Actually I oppose just a little bit after all. I mean if I merge them into `AccepError` then it'd like reusing the same error type for a completely different operation, which kind of hurts my head.

I would not focus on that they have error codes with the same name but to the fact that they're responses to completely different operations. Imagine as they were just empty structs.

If I merge them I will have to rename the enum as `SecInfoError`. That way it is at least somehow consistent. @npmccallum ?

Reply:
I.e. they are not symmetrical operations even tho they have similar names.